pkg/controllers/vdb/revivedb_reconciler.go

```diff
@@ -96,6 +101,18 @@ func (r *ReviveDBReconciler) Reconcile(ctx context.Context, _ *ctrl.Request) (ct
 	return g.checkAndRunInit(ctx)
 }
 
+func (r *ReviveDBReconciler) isRestoreSupported() error {
+	if vinf, ok := r.Vdb.MakeVersionInfo(); !vmeta.UseVClusterOps(r.Vdb.Annotations) ||
+		!ok || !vinf.IsNewer(vapi.RestoreUnsupportedMaxVersion) {
+		r.VRec.Eventf(r.Vdb, corev1.EventTypeWarning, events.ReviveDBRestoreUnsupported,
+			fmt.Sprintf("Restoring from a restore point is unsupported in ReviveDB given the current server version and deployment method, "+
```

Comment:
If we are generating the same error message for both, lets generate it once and use it as a string. We can use `VRec.Event` and `errors.New` if the message is already created.